### PR TITLE
dev 16

### PR DESCRIPTION
**Pull request by TsaiMengXuan**

### CODE REVIEW

File: app/api/demand_group_setting.py [hunk 1]
```diff
@@ -94,51 +99,71 @@ def demand_group_info_setting():
...
+        unload_mode = request.form.get('unload-mode', default=0, type=int)
+        
+        app_unload_group = AppUnloadGroup(
+            unload_group_sn=unload_group_sn,
+            group_name = group_name,
+            sort = sort,
+            gateway_sn=GATEWAY_SN,
+            unload_mode=unload_mode,
+            group_num=unload_group_sn,
+            group_status=50
```
Comment (RobinChien) [inline app/api/demand_group_setting.py:148]: Why 50?

File: app/api/group.py [hunk 2]
```diff
@@ -211,7 +206,22 @@ def add_group():
...
+    # 例如：若在同一個 gateway 與同一個 area，會有一個 list，group_sn 從 1 開始
+    # group_num 應該是要計算在同一台 gateway 上，總共有幾個群組
+    # 綜上所述，判定只要在同一個 gateway、area 上增加群組，會在這個 list 的 group_sn 與 group_num 都加 1
+    # request 會用到 group_name、gateway_sn、area_sn
+    # group_id 似乎就是 group_sn
+    
+    # pseudo_code
+        # from setting import GATEWAY_SN, area_sn
+    # 但是 setting.py 沒有將 area_sn 設定成全域變數
+    # area_sn = request.form.get('area_sn', default = '', type = int)
```
Comment (RobinChien) [inline app/api/group.py:223]: remove

File: app/api/group.py [hunk 3]
```diff
@@ -220,6 +230,15 @@ def add_group():
...
         group_sn = lib_generate_id.generate_id(3, session['gateway-id'])
+        
+    # 找到同一個 gateway、area 裡面的 MAX(group_sn)，並 + 1
+    # max_group_sn_sql = ("SELECT MAX(`group_sn`) as `max_group_sn`"
+    #                    "FROM `app_group` "
+    #                    "WHERE `gateway_sn` = '{}' AND `area_sn` = `{}` "
+    #                      ).format(GATEWAY_SN, area_sn)
+    # dbh = MySQL()
+    # max_group_sn = dbh.query(max_group_sn_sql)[0].get('max_group_sn')
+    # group_sn = max_group_sn + 1
```
Comment (RobinChien) [inline app/api/group.py:241]: remove

File: app/api/group.py [hunk 4]
```diff
@@ -339,14 +358,23 @@ def delete_group():
...
+        
+        # app_group = AppGroup(
+        #     gateway_sn = GATEWAY_SN,
+        #     group_sn = group_sn,
+        #     group_number = group_number, 
+        #     group_name = group_name
+        # )
+        
+        # delete_group_node_sql = "DELETE FROM `group_node_list` WHERE `gateway_sn` = {} AND `group_sn` = {}".format(
+        #    session['gateway-id'], group_sn)
```
Comment (RobinChien) [inline app/api/group.py:373]: remove

File: app/api/group.py [hunk 5]
```diff
@@ -390,6 +419,10 @@ def delete_group():
 
         if not result:
             raise Exception("Insert mqtt_message Fail.")
+        
+        # delete_group_nodes_sql = "DELETE FROM `group_node_list` WHERE (`gateway_sn`, `group_id`) VALUES {}".format(GATEWAY_SN, group_sn)
+        # delete_group_nodes = db.engine.execute(delete_group_nodes_sql)
+        # db.session.close()
```
Comment (RobinChien) [inline app/api/group.py:425]: remove

File: app/utils/database_util.py [hunk 6]
```diff
@@ -1190,7 +1190,46 @@ def select_app_scene_node(gateway_sn: int, scene_sn: int):
...
+#         print(result)
+#         dbh.close()
+
+#         if not result:
+#             raise Exception(get_function_name(), " Fail")
+
+#         return result
+#     except Exception as e:
+#         trace_error(e)
+#         return False
```
Comment (RobinChien) [inline app/utils/database_util.py:1210]: remove

File: app/api/area.py [hunk 7]
```diff
@@ -224,7 +224,8 @@ def area_delete():
             area_name=area_name,
             gateway_sn=GATEWAY_SN
         )
-
+        
+        # delete_app_area 模組不見
```
Comment (RobinChien) [inline app/api/area.py:228]: remove

Comment (TsaiMengXuan) [reply app/api/area.py:228]: done

File: app/utils/database_util.py (hunk 6 above)
Comment (TsaiMengXuan) [reply app/utils/database_util.py:1210]: done

File: app/api/group.py (hunk 4 above)
Comment (TsaiMengXuan) [reply app/api/group.py:373]: done

File: app/api/group.py (hunk 3 above)
Comment (TsaiMengXuan) [reply app/api/group.py:241]: done

File: app/api/group.py (hunk 2 above)
Comment (TsaiMengXuan) [reply app/api/group.py:223]: done

**Pull request merged**